### PR TITLE
Filebeat azure module pipeline fixes and changes

### PR DESCRIPTION
**Pull request by narph**
## What does this PR do?

- fixes several issues and adds more parsing options
- fixes escape in the logs
- adds more parsing options for resource id
- fixes mapping issues for json fields
- fixes parsing issues with IP

## Why is it important?

- fixes several issues and adds more parsing options

## Checklist

- [x] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Related issues
- Closes https://github.com/elastic/beats/issues/25815, https://github.com/elastic/beats/issues/24392, https://github.com/elastic/beats/issues/20797, https://github.com/elastic/beats/issues/25127, https://github.com/elastic/beats/issues/24292

### CODE REVIEW

File: x-pack/filebeat/module/azure/azure-shared-pipeline.yml [hunk 1]
```diff
@@ -7,6 +7,7 @@ processors:
     field: azure.resource_id
     patterns:
     - /SUBSCRIPTIONS/%{SUBID:azure.subscription_id}/RESOURCEGROUPS/%{GROUPID:azure.resource.group}/PROVIDERS/%{PROVIDERNAME:azure.resource.provider}/NAMESPACES/%{NAMESPACE:azure.resource.namespace}/AUTHORIZATIONRULES/%{RULE:azure.resource.authorization_rule}
+    - /subscription/%{SUBID:azure.subscription_id}/resourceGroups/%{GROUPID:azure.resource.group}/providers/%{PROVIDERNAME:azure.resource.provider}/namespaces/%{NAMESPACE:azure.resource.namespace}/authorizationRules/%{RULE:azure.resource.authorization_rule}
```
Comment (jsoriano) [inline x-pack/filebeat/module/azure/azure-shared-pipeline.yml:10]: For testing, do we have any example log file that uses these lowercased forms?

File: x-pack/filebeat/module/azure/azure-shared-pipeline.yml [hunk 2]
```diff
@@ -16,21 +17,66 @@ processors:
...
+    pattern_definitions:
+      SUBID: (\{){0,1}[0-9a-fA-F]{8}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{12}(\}){0,1}
+      GROUPID: .+
+    ignore_failure: true
+- grok:
+    field: azure.resource_id
+    if: 'ctx.azure?.subscription_id == null'
+    patterns:
+      - /SUBSCRIPTIONS/%{SUBID:azure.subscription_id}
+      - /subscriptions/%{SUBID:azure.subscription_id}
```
Comment (jsoriano) [inline x-pack/filebeat/module/azure/azure-shared-pipeline.yml:76]: Grok doesn't have case insensitive patterns, right? :confused:

File: x-pack/filebeat/module/azure/platformlogs/test/platformlogs-eventhub.log [hunk 3]
```diff
@@ -1 +1 @@
-{"ActivityId":"30ed877c-a36b-491a-bd4d-ddd847fe55b8","Caller":"Portal","Environment":"PROD","EventName":"Retreive ConsumerGroup","EventProperties":"{\"SubscriptionId\":\"7657426d-c4c3-44ac-88a2-3b2cd59e6dba\",\"Namespace\":\"obstesteventhubs\",\"Via\":\"sb://obstesteventhubs.servicebus.windows.net/insights-logs-operationallogs/consumergroups?api-version=2017-04\u0026$skip=0\u0026$top=100\",\"TrackingId\":\"30ed877c-a36b-491a-bd4d-ddd847fe55b8_M2CH3_M2CH3_G3S2\"}","EventTimeString":"11/3/2020 9:06:42 AM +00:00","Region":"West Europe","ScaleUnit":"PROD-AM3-AZ501","Status":"Succeeded","category":"OperationalLogs","resourceId":"/SUBSCRIPTIONS/7657426D-C4C3-44AC-88A2-3B2CD59E6DBA/RESOURCEGROUPS/OBS-TEST/PROVIDERS/MICROSOFT.EVENTHUB/NAMESPACES/OBSTESTEVENTHUBS"}
```
Comment (jsoriano) [inline x-pack/filebeat/module/azure/platformlogs/test/platformlogs-eventhub.log:1]: Why removing this test log line? Wasn't it valid?

File: x-pack/filebeat/module/azure/platformlogs/test/platformlogs-kube.log [hunk 4]
```diff
@@ -1 +1 @@
-{"Cloud":"AzureCloud","Environment":"prod","category":"kube-audit","ccpNamespace":"5e4bf4baee195b00017cdbfa","operationName":"Microsoft.ContainerService/managedClusters/diagnosticLogs/Read","properties":{"log":"{\"kind\":\"Event\",\"apiVersion\":\"audit.k8s.io/v1\",\"level\":\"Metadata\",\"auditID\":\"22af12c3-a1fe-4f2c-99a9-3cdde671dbfe\"}","pod":"kube-apiserver-666bd4b459-hjgdc","stream":"stdout"},"resourceId":"/SUBSCRIPTIONS/70BD6E77-4B1E-4835-8896-DB77B8EEF364/RESOURCEGROUPS/OBS-INFRASTRUCTURE/PROVIDERS/MICROSOFT.CONTAINERSERVICE/MANAGEDCLUSTERS/OBSKUBE","time":"2020-11-09T10:57:31.0000000Z"}
```
Comment (jsoriano) [inline x-pack/filebeat/module/azure/platformlogs/test/platformlogs-kube.log:1]: Also here, why removing this test case?

**Review by jsoriano — COMMENTED** [review]
Looks good, only a couple of small things.

> * Closes #25815, #24392, #20797, #25127, #24292

:rocket:

File: x-pack/filebeat/module/azure/azure-shared-pipeline.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (narph) [reply x-pack/filebeat/module/azure/azure-shared-pipeline.yml]: fixed

File: x-pack/filebeat/module/azure/platformlogs/test/platformlogs-eventhub.log (hunk 3 above)
Comment (narph) [reply x-pack/filebeat/module/azure/platformlogs/test/platformlogs-eventhub.log:1]: simplified the use case, I've changed the mapping so I would have had to add it for those extra fields

File: x-pack/filebeat/module/azure/azure-shared-pipeline.yml (hunk 2 above)
Comment (narph) [reply x-pack/filebeat/module/azure/azure-shared-pipeline.yml:76]: I think so, `(?i)` but I had troubles making it work with `resourceGroups`, could have been on my side

File: x-pack/filebeat/module/azure/azure-shared-pipeline.yml (hunk 1 above)
Comment (narph) [reply x-pack/filebeat/module/azure/azure-shared-pipeline.yml]: we do now in the edge case file

**Review by jsoriano — APPROVED** [review]
:+1:

**Pull request merged**